include/zephyr/drivers/spi.h

```diff
@@ -242,19 +242,38 @@ extern "C" {
  *
  */
 struct spi_cs_control {
-	/**
-	 * GPIO devicetree specification of CS GPIO.
-	 * The device pointer can be set to NULL to fully inhibit CS control if
-	 * necessary. The GPIO flags GPIO_ACTIVE_LOW/GPIO_ACTIVE_HIGH should be
-	 * equivalent to SPI_CS_ACTIVE_HIGH/SPI_CS_ACTIVE_LOW options in struct
-	 * spi_config.
-	 */
-	struct gpio_dt_spec gpio;
-	/**
-	 * Delay in microseconds to wait before starting the
-	 * transmission and before releasing the CS line.
-	 */
-	uint32_t delay;
+	union {
+		struct {
+			/**
+			 * GPIO devicetree specification of CS GPIO.
+			 * The device pointer can be set to NULL to fully inhibit CS control if
+			 * necessary. The GPIO flags GPIO_ACTIVE_LOW/GPIO_ACTIVE_HIGH should be
+			 * equivalent to SPI_CS_ACTIVE_HIGH/SPI_CS_ACTIVE_LOW options in struct
+			 * spi_config.
+			 */
+			struct gpio_dt_spec gpio;
+			/**
+			 * Delay in microseconds to wait before starting the
+			 * transmission and before releasing the CS line.
+			 */
+			uint32_t delay;
+		};
+		struct {
+			/**
+			 * CS enable lead time, i.e. how long should the CS be asserted
+			 * before the first clock. Specified in nanoseconds.
+			 */
+			uint32_t lead;
```

Comment:
Why not reusing same terminology than the new bindings ?
s/lead/cs_setup
s/lag/cs_hold

Introducing new terminology makes things confusing IMO, and will complicate searching through the code base.

Reply:
There was not old terminology because this was never supported, not sure what you're talking about

Reply:
I mean terminology used in the device tree bindings (hold-on, set-up)

Reply:
I think  @erwango has a point here, setup_delay/hold_delay are indeed more self-documented than lead/lag 
(plus they directly relate to their respective dts  counterparts)